modules/lang-painless/src/main/java/org/elasticsearch/painless/api/Augmentation.java

```diff
@@ -552,4 +554,138 @@ public static String decodeBase64(String receiver) {
+        return getByPath(receiver, splitPath(path), 0, getByPathMissing(path));
+    }
+
+    /**
+     * Same as 'getByPath(List, String)' but returns defaultValue if path does not exist.
+     * Throws 'IllegalArgumentException' if there is a non-container type at a non-terminal path element.
+     */
+    public static <E> Object getByPath(List<E> receiver, String path, Object defaultValue) {
+        return getByPath(receiver, splitPath(path), 0, () -> defaultValue);
+    }
+
+    /**
+     * Same as 'getByPath(Map, String)' but returns defaultValue if path does not exist.
+     * Throws 'IllegalArgumentException' if there is a non-container type at a non-terminal path element.
+     */
+    public static <K,V> Object getByPath(Map<K,V> receiver, String path, Object defaultValue) {
+        return getByPath(receiver, splitPath(path), 0, () -> defaultValue);
+    }
+
+    // getByPath helpers
+    /**
+     * Dispatch function for getByPath, handling successful lookup, obj as Map, List or non-collection.
+     * Throws 'IllegalArgumentException' if elements[i] is an empty string, indicating double separators
+     * or path ends in a separator.
+     */
+    private static Object getByPath(Object obj, String[] elements, int i, Supplier<Object> defaultSupplier) {
+        if (i > elements.length - 1) {
+            return obj;
+        } else if (elements[i].length() == 0 ) {
+            String format = "Extra '.' in path [%s] at index [%d]";
+            throw new IllegalArgumentException(String.format(Locale.ROOT, format, String.join(".", elements), i));
+        } else if (obj instanceof Map<?,?>) {
+            return getByPath((Map<?,?>) obj, elements, i, defaultSupplier);
+        } else if (obj instanceof List<?>) {
+            return getByPath((List<?>) obj, elements, i, defaultSupplier);
+        }
+        return handleMissing(obj, elements, i, defaultSupplier);
+    }
+
+    /**
+     * Map version of getByPath, calls dispatch function on next path element with object map[elements[i]] or
+     * return defaultSupplier if element is missing an i is the last index of elements.
+     * Throws 'IllegalArgumentException' is element is missing and i is not the last index of elements.
+     */
+    private static <K,V> Object getByPath(Map<K,V> map, String[] elements, int i, Supplier<Object> defaultSupplier) {
+        String element = elements[i];
+        if (map.containsKey(element)) {
+            return getByPath(map.get(element), elements, i + 1, defaultSupplier);
+        }
+        return handleMissing(map, elements, i, defaultSupplier);
+    }
+
+
+    /**
+     * List version of getByPath, calls dispatch function on next path element with object list[elements[i]] or
+     * return defaultSupplier if element is missing an i is the last index of elements.
+     * Throws 'IllegalArgumentException' is element is missing and i is not the last index of elements.
```

Comment:
You can use `@throws` instead of prose here instead of text. Same for `@return` above

Reply:
Updated to use javadoc `@throws` and `@return`.